scipy/linalg/_special_matrices.py

```diff
@@ -1396,3 +1398,339 @@ def convolution_matrix(a, n, mode='full'):
         col0 = az
         row0 = raz[-n:]
     return toeplitz(col0, row0)
+
+
+def laplacian_Nd(
+    shape, bc="N", form="sparse", return_evals=False, dtype="float"
+):
+    """
+    Construct a sparse matrix and a callable associated with the Laplacian
+    on a uniform rectangular grid in `N` dimensions given by tuple `shape`
+    and output its eigenvalues.
+
+    The Laplacian matrix `L` is square real symmetric negative definite.
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the grid for the `N` dimensional Laplacian of length `N`
+        where the shape `n` of the Laplacian matrix `L` is ``np.prod(shape)``
```

Comment:
Having both `N` and `n` here is confusing. Also "`N` dimensional Laplacian of length `N`" can be misread.

How about something like:
A tuple of length `N` (corresponding to the dimension of the Lapacian), where
each entry gives the size of that dimension. The Laplacian matrix `L` will be a
square matrix of size ``np.prod(shape)``.

Reply:
Done